modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayEnvironmentsAdministrator.testcase

```diff
@@ -18,47 +18,93 @@ definition {
 	@priority = "5"
-	test AdminCanAddAndDeleteFactorCategory {
+	test AdminCanAddAndDeleteFactorOption {
+		task ("Go to Categories on Manage Sidebar and Add Category") {
+			ApplicationsMenu.gotoSite(site = "Testray 2");
 
-		// TODO LPS-157239 AdminCanAddAndDeleteFactorCategory pending implementation
+			TestrayLayout.clickManageOnSidebar();
 
-	}
+			TestrayManage.selectOptionsManage(itemValue = "Categories");
 
-	@description = "This is a test for LPS-157327 - Verify if As an Administrator I should be able to add and delete a Factor Option"
-	@ignore = "Test Stub"
-	@priority = "5"
-	test AdminCanAddAndDeleteFactorOption {
+			TestrayManage.addNewCategory(category = "Environments_2 Factor Category");
+		}
+
+		task ("Go to Options on Manage Sidebar and Add New Option") {
+			TestrayLayout.clickManageOnSidebar();
+
+			TestrayManage.selectOptionsManage(itemValue = "Options");
+
+			TestrayManage.createNewOption(
+				categorie = "Environments_2 Factor Category",
+				key_value = "Environments_2 Factor Option");
+		}
+
+		task ("Delete Option, Go to Options and state if deleted") {
+			TestrayManage.deleteOption(nameOptions = "Environments_2 Factor Option");
 
-		// TODO LPS-157327 AdminCanAddAndDeleteFactorOption pending implementation
+			TestrayManage.deleteCategory();
 
+			TestrayManage.assertOptionNotPresent(key_title = "Environments_2 Factor Option");
+
+			TestrayManage.assertNameNotPresent(name = "Environments_2 Factor Category");
+		}
 	}
 
 	@description = "This is a test for LPS-157231 - Verify if as an Administrator I am able to add Factor Categories"
-	@ignore = "Test Stub"
 	@priority = "5"
 	test AdminCanAddCategory {
+		task ("Go to Categories on Manage Sidebar") {
+			ApplicationsMenu.gotoSite(site = "Testray 2");
+
+			TestrayLayout.clickManageOnSidebar();
+
+			TestrayManage.selectOptionsManage(itemValue = "Categories");
+		}
 
-		// TODO LPS-157231 AdminCanAddCategory pending implementation
+		task ("Add a category and check if the category was created") {
+			TestrayManage.addNewCategory(category = "Environments_1 Category");
 
+			TestrayManage.assertName(name = "Environments_1 Category");
+		}
 	}
 
 	@description = "This is a test for LPS-157309 - Verify if as an Administrator I am able to add Options"
-	@ignore = "Test Stub"
 	@priority = "5"
 	test AdminCanAddOptions {
+		task ("Go to Categories on Manage Sidebar and Add Category") {
+			ApplicationsMenu.gotoSite(site = "Testray 2");
+
+			TestrayLayout.clickManageOnSidebar();
+
+			TestrayManage.selectOptionsManage(itemValue = "Categories");
 
-		// TODO LPS-157309 AdminCanAddOptions pending implementation
+			TestrayManage.addNewCategory(category = "Environments_2 Category");
+		}
+
+		task ("Go to Options on Manage Sidebar, Add Options and assert the new option has been created") {
+			TestrayLayout.clickManageOnSidebar();
+
+			TestrayManage.selectOptionsManage(itemValue = "Options");
+
+			TestrayManage.createNewOption(
+				categorie = "Environments_2 Category",
```

Comment:
Hi @joelrocha68 would you please correct this word? "categorie" must be changed to "category"